src/js/worklets/chat/messages.js

```diff
@@ -0,0 +1,67 @@
+import { useDerivedValue, withTiming, interpolate } from 'react-native-reanimated';
+
+export function navigationHeaderOpacity(bigNameVisible) {
+  return useDerivedValue(function () {
+    'worklet';
+    return bigNameVisible.value ? withTiming(0) : withTiming(1);
+  });
+}
+
+export function navigationHeaderPosition(bigNameVisible, topBarHeight) {
+  return useDerivedValue(function () {
+    'worklet';
+    return bigNameVisible.value ? withTiming(topBarHeight) : withTiming(0);
+  });
+}
+
+const defaultExtrapolationOption = {
+  extrapolateLeft: 'clamp',
+  extrapolateRight: 'clamp',
+};
+
+export function navigationBackgroundOpacity(props) {
+  return useDerivedValue(function () {
+    'worklet';
+    const {
+      'distance-from-list-top': distanceFromListTop,
+      'header-container-top-margin': headerContainerTopMargin,
+      'top-bar-height': topBarHeight,
+      'all-loaded?': isAllLoaded,
+    } = props;
+    if (isAllLoaded.value) {
+      return interpolate(
+        distanceFromListTop.value,
+        [headerContainerTopMargin, headerContainerTopMargin + topBarHeight],
+        [0, 1],
+        defaultExtrapolationOption,
+      );
+    } else {
+      return 1;
+    }
+  });
+}
+
+export function navigationPinnedBannerOpacity(props) {
+  return useDerivedValue(function () {
+    'worklet';
+    const {
+      'distance-from-list-top': distanceFromListTop,
+      'top-bar-height': topBarHeight,
+      'navigation-view-height': navigationViewHeight,
+      'all-loaded?': isAllLoaded,
+    } = props;
+
+    if (isAllLoaded.value) {
+      return interpolate(
+        distanceFromListTop.value,
+        // 142 - Position of pinned banner for starting animation (currently decided as per hit and trial)
+        // exact position needs to confirmed with design team - https://github.com/status-im/status-mobile/issues/18132
+        [navigationViewHeight + 142, navigationViewHeight + 142 + topBarHeight],
```

Comment:
I guess before the exact number is confirmed with the design team, it could be extracted and named inside cljs, so it's clearer what it represents.